File: 2028/Solution.js
/**
 * @param {number[]} rolls
 * @param {number} mean
 * @param {number} n
 * @return {number[]}
 */
var missingRolls = function(rolls, mean, n) {
    let dif = (n + rolls.length) * mean;
    for(let i of rolls){
        dif -= i;
    }
    if(dif < n || dif > n * 6) {
        return [];
    }
    let res = new Array(n).fill(Math.floor(dif/n));
    let r = dif % n;
    for(let i = 0; i < r; i++){
        res[i]++;
    }
    return res;
};

File: 2028/Solution.go
func missingRolls(rolls []int, mean int, n int) []int {
    dif := (n + len(rolls)) * mean
    for _, x := range rolls {
        dif -= x
    }
    if dif < n || dif > n * 6 {
        return []int{}
    }
    res := make([]int, n)
    d := dif / n
    for i, _ := range res {
        res[i] = d
    }
    for i := 0; i < dif % n; i++ {
        res[i]++
    }
    return res
}
